### PR TITLE
Adds tier 4 stock parts

### PR DESCRIPTION
**Pull request by MadmanMartian**
Still thinking on the method of acquisition. High tier vault loot perhaps?

:cl:
 * rscadd: Recent excavations of the asteroid have yielded findings of an advanced civilization. Xenoarchs are recommended to look for parts that are compatible with current machinery.

### CODE REVIEW

**Comment by DrSnips** [comment]
_You hear a maddening whisper in your head "protolathe"_
Xenoarch and ROID vault stock parts are currently garbage and pointless because you can make all of then in the protolathe. I'd say they could find a home there if anywhere. Please allow them to be mechanic scannable at the least.
As for an non-meme protolathe option items that are unlocked with anomaly research are pretty limited so it could go there. If you do go with this way leave bluespace bins as is.

**Comment by Blithering** [comment]
Maybe have them salvaged from an away mission if you aren't going with standard construction.

**Comment by DrSnips** [comment]
I don't think away missions are the right place for them. Stock parts are intended to be used in rounds where the main focus is the station and interactions within it such as medical, science, or any thing else really that uses stock parts to improve efficiency as the round progresses. When an away mission happens people aren't focusing on the station anymore, or are even on it for that matter, which would result in them having no use for the places that do use them.

**Comment by boy2mantwicethefam** [comment]
Protolathe stuff. I was thinking make 4th tier require diamonds and 5th tier require phazon

**Comment by SonixApache** [comment]
The roid already has random small rooms that have tier3 parts in them. Just throw them in there.

For the love of god, don't make it more lathecreep.

**Comment by ghost** [comment]
Nobody will care for these parts if they cannot be made in protolathe by doing research. Putting them behind current tier 3 + anomaly research would make them sometimes used.

These parts offer negligible benefits, a bit of saved time and a bit of saved materials for printing. The saved time will be spent getting the research for them. Saved materials will be irrelevant regardless of effort spent, since science is a binary department. It either has no materials or miners bring so much materials that you can build another station.

**Comment by iFlashYou** [comment]
Just put it into the protolathe and make it cost diamonds and phazon. 
I do not know why some people are afraid of the protolathe, everything new that gets added, especially science stuff, some people try it with force to get it from somewhere else and cry "muh lathecreep" when just someone thinks about putting something new into the protolathe.

**Comment by SNilWillIllShill** [comment]
Xenoarch tech site stock parts are almost always tier 1. You might as well put these tier 4s in tech sites instead, make them worth going for in a round - or have them backed by anomaly research as well to incentivise xenoarchs bringing back small junk artifacts to destructive analyzer for the levels required. @codeungabunga is right, there are other ways to have them in and able to be obtained without resorting to this.

File: code/modules/research/xenoarchaeology/finds/finds.dm [hunk 1]
```diff
@@ -500,13 +500,18 @@
...
-/datum/find/stock_parts
+/datum/find/stock_parts //Tier 4 parts
 	find_ID = ARCHAEO_STOCKPARTS
 	apply_material_decorations = FALSE
 	anomaly_factor = 2
 
 /datum/find/stock_parts/spawn_item()
-	var/list/possible_spawns = existing_typesof(/obj/item/weapon/stock_parts)
+	var/list/possible_spawns = list(
+			/obj/item/weapon/stock_parts/capacitor/adv/super/ultra,
```
Comment (jknpj) [inline code/modules/research/xenoarchaeology/finds/finds.dm:510]: could make a global list with these but eh

**Comment by SonixApache** [comment]
update OP, including 🆑

**Comment by MadmanMartian** [comment]
it dun

**Pull request merged**